src/NUnitEngine/nunit.engine/Internal/ServerUtilities.cs

```diff
@@ -69,16 +70,22 @@ private static TcpChannel CreateTcpChannel( string name, int port, int limit )
             BinaryClientFormatterSinkProvider clientProvider =
                 new BinaryClientFormatterSinkProvider();
 
-            return new TcpChannel( props, clientProvider, serverProvider );
+            return new TcpChannel(
+                props,
+                clientProvider,
+                currentMessageCounter != null
+                    ? new ObservableServerChannelSinkProvider(currentMessageCounter) { Next = serverProvider }
+                    : (IServerChannelSinkProvider)serverProvider);
         }
 
         /// <summary>
         /// Get a default channel. If one does not exist, then one is created and registered.
         /// </summary>
+        /// <param name="currentMessageCounter">An optional counter to provide the ability to wait for all current messages</param>
         /// <returns>The specified <see cref="TcpChannel"/> or null if it cannot be found and created</returns>
-        public static TcpChannel GetTcpChannel()
+        public static TcpChannel GetTcpChannel( CurrentMessageCounter currentMessageCounter = null )
         {
-            return GetTcpChannel( "", 0, 2 );
+            return GetTcpChannel( "", 0, 2, currentMessageCounter );
```

Comment:
I believe this is dead code now, but I kept running into so much dead code around remoting that it seems best to leave it for a later overhaul.